nat-lab/tests/test_vpn.py

```diff
@@ -104,43 +117,75 @@ async def _connect_vpn(
                 adapter_type=AdapterType.BoringTun,
                 connection_tracker_config=generate_connection_tracker_config(
                     ConnectionTag.MAC_VM,
-                    vpn_1_limits=ConnectionLimits(1, 1),
                     stun_limits=ConnectionLimits(1, 1),
                 ),
                 is_meshnet=False,
             ),
             "10.0.254.7",
-            marks=[
-                pytest.mark.mac,
-            ],
+            marks=pytest.mark.mac,
         ),
     ],
 )
+@pytest.mark.parametrize(
+    "vpn_conf",
+    [
+        pytest.param(VpnConfig(config.WG_SERVER, ConnectionTag.DOCKER_VPN_1, True), id="wg_server"),
+        pytest.param(VpnConfig(config.NLX_SERVER, ConnectionTag.DOCKER_NLX_1, False), id="nlx_server"),
+    ],
+)
 async def test_vpn_connection(
-    alpha_setup_params: SetupParameters, public_ip: str
+    alpha_setup_params: SetupParameters,
+    vpn_conf: VpnConfig,
+    public_ip: str,
 ) -> None:
+    def find_cfg_by_key(
+        params: SetupParameters, key: str
+    ) -> Optional[ConnectionTrackerConfig]:
+        if params is not None and params.connection_tracker_config is not None:
+            for cfg in params.connection_tracker_config:
+                if cfg.get_key() == key:
+                    return cfg
+        return None
+
     async with AsyncExitStack() as exit_stack:
+        alpha_setup_params = copy.deepcopy(alpha_setup_params)
+
+        if vpn_conf.conn_tag == ConnectionTag.DOCKER_VPN_1:
+            cfg = find_cfg_by_key(alpha_setup_params, "vpn_1")
+            assert cfg is not None
+            cfg.limits = ConnectionLimits(1, 1)
```

Comment:
does it actually write to `alpha_setup_params`

Reply:
Yep, it actually overwrites the original, that's why there's deepcopy above it. I caught it redhanded when it was failing with both tests but succeeded with each one separately